runtime/include/chpl-comm-native-atomics.h

```diff
@@ -0,0 +1,168 @@
+/*
+ * Copyright 2004-2018 Cray Inc.
+ * Other additional copyright holders may be indicated within.
+ *
+ * The entirety of this work is licensed under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ *
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef _chpl_comm_native_atomics_h_
+#define _chpl_comm_native_atomics_h_
+
+//
+// These are the declarations for network-native atomic ops, for the
+// comm layers that support those.
+//
+
+//
+// We support 32- and 64-bit signed ints, uints and reals, although we
+// don't necessarily support all of these types for all operations.
+//
+
+//
+// Do a remote atomic store.  The value to be stored is *desired on the
+// local node.  The target location to be stored into is *object on the
+// given node.  This differs from a regular chpl_comm_put() in that it
+// is coherent with other chpl_comm_atomic_*() operations (but may incur
+// network overhead) even if the target is the calling node.
+//
+#define DECL_CHPL_COMM_ATOMIC_PUT(type)                                 \
```

Comment:
Maybe not for this PR, but we should change ATOMIC_{PUT,GET} to ATOMIC_{STORE,LOAD} (or ATOMIC_{WRITE,READ}. PUT/GET is ugni implementation detail IMO.

Reply:
Yes, good idea.  See https://github.com/chapel-lang/chapel/issues/11653.